components/NoteEditor/Editor/Heading.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useRef, useEffect } from 'react';
+import styles from './Editor.module.scss';
+
+export default function Heading({
+  name, placeholder, value, setValue,
+}) {
+  const el = useRef();
+  const [currentHeight, setCurrentHeight] = useState();
+
+  const removeNewLine = (text) => text.replace(/\n/g, '');
+
+  const handleChange = (e) => {
+    const text = removeNewLine(e.currentTarget.value);
+
+    setCurrentHeight(0);
+    setValue(text);
+  };
+
+  useEffect(() => {
+    setCurrentHeight(el.current.scrollHeight);
+  }, []);
```

Comment:
You could probably just set the initial currentHeight state to el.current.scrollHeight, so you wouldnt need this

Reply:
I can't because the `ref` value is only set after the HTML renders, before the state is declared

Reply:
fair enough